docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,264 @@
+Unzipping `peptipedia_csv.zip` results in a CSV file `peptipedia.csv` that records the peptide id, peptide sequence, and predicted bioactivity labels.
+Unzipping `backup_sql.zip` results in a postgres SQL dump file `backup_dump.sql`.
+This file contains metadata associated with the peptide sequences.
+To extract the metadata to separate CSV tables, we used the python code:
+```
+import re
+
+output_file_prefix = 'table_'
+current_table = None
+output_file = None
+
+with open('backup_dump.sql', 'r') as dump_file:
+    for line in dump_file:
+        # Check for COPY command start
+        if line.startswith('COPY'):
+            table_name = line.split()[1]  # Assumes table name is the second word
+            current_table = table_name
+            output_file = open(f'{output_file_prefix}{current_table}.csv', 'w')
+        elif line.startswith('\.'):
+            # End of data for the current table
+            if output_file:
+                output_file.close()
+                output_file = None
+        elif output_file:
+            # Convert tab-separated values to comma-separated values
+            csv_line = line.replace('\t', ',')
+            output_file.write(csv_line)
+```
+
+4. Filter to peptides that are at least k in length. MMSeqs2 clustering in step 4 will fail if any of the sequences are below the k-mer length used for clustering.
+```
+seqkit seq --min-len 10 -o peptipedia_minlen10.fasta peptipedia.fasta
+```
+
+5. Cluster the sequences.
+Using a low identity threshold (30%), this step determines which peptides in the input database have similar sequences.
+We will subsequently use this information to artificially generate a data set where the peptides or their characteristics are predictive of bioactivity labels.
+```
+mmseqs easy-cluster peptipedia_minlen10.fasta mmseqs/peptipedia_minlen10-0.3 tmp --min-seq-id 0.3
+```
+
+6. Filter to large clusters composed of peptides that all have the same bioactivity prediction and write a TSV file.
+Many of the peptides that end up in clusters have multiple bioactivity predictions (e.g. antimicrobial and therapeutic).
+The script below filters to peptide clusters that all have at least one bioactivity label that is the same.
+It also adds additional metadata (GO, PFAM) from the SQL database to the data.frame of peptide information.
+```
+library(tidyverse)
+library(janitor)
+
+# The goal of this script is to create a demo machine learning data set from peptides and their metadata.
+# It filters peptides in the peptipedia database to clusters of similar peptides
+# (30% identity or greater) that all have the same bioactivity label.
+# It then joins selected peptides to their metadata
+
+
+# filter to peptides clusters with many members that have the same bioactivity --------
+
+# read in cluster information from mmseqs2 linclust
+clusters <- read_tsv("~/Downloads/backup_peptipedia_29_03_2023/mmseqs/peptipedia_minlen10-0.3_cluster.tsv",
```

Comment:
is this the result of the previous step? (not explicitly mentioned, but maybe I'm missing it)

Reply:
yes! I added a comment and fixed the tool name to make this more clear.